File: 0050-powx-n/0050-powx-n.rb
# @param {Float} x
# @param {Integer} n
# @return {Float}
def my_pow(x, n)
    return 1 if n == 0
    if n > 0
        while n > 0
            return x * my_pow(x, n-1) if n % 2 > 0
            return my_pow(x*x, n/2)
        end
    else
        while n < 0
            return my_pow(x, n+1) / x if n % 2 > 0
            return my_pow(x*x, n/2)
        end
    end
end

File: 0050-powx-n/0050-powx-n.ts
function myPow(x: number, n: number): number {
  if (n === 0) {
    return 1
  } else if (n > 0) {
    return n % 2 ? x * myPow(x, n - 1) : myPow(x * x, n / 2)
  } else if (n < 0) {
    return n % 2 < 0 ? myPow(x, n + 1) / x : myPow(x * x, n / 2)
  }
}